Dalamud/Plugin/Services/IClientState.cs

```diff
@@ -9,11 +9,35 @@ namespace Dalamud.Plugin.Services;
 /// </summary>
 public interface IClientState
 {
+    /// <summary>
+    /// A delegate type used for the <see cref="ClassJobChanged"/> event.
+    /// </summary>
+    /// <param name="classJobId">The new ClassJob id.</param>
+    public delegate void ClassJobChangeDelegate(uint classJobId);
```

Comment:
Would it make sense to expose `LazyRow<ClassJob>` here (and elsewhere) instead for convenience? I don't have an objection to just the `uint`, but it's probably worth asking.

Reply:
I actually didn't think of that, because `TerritoryChanged` event also just passes an `ushort`.
If you want me to add it to the new events, sure.

Reply:
I don't really have a good answer atm if it'd be "more appropriate" to return the rich type or the plain ID. One makes it easier to grab related data (although in almost all cases, the plugin dev will already be able to resolve their own sheet), and the other would ensure maximum compatibility. 

We can always change this later I suppose, so I don't think there's any harm in leaving it as ID-only. I'll leave this open for a little bit though to let anyone with Strong Opinions:tm: make their feelings known.